src/main/java/com/transferwise/openbanking/client/api/common/ExceptionUtils.java

```diff
@@ -0,0 +1,25 @@
+package com.transferwise.openbanking.client.api.common;
+
+import com.transferwise.openbanking.client.error.ApiCallException;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.reactive.function.client.WebClientException;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+@UtilityClass
+@Slf4j
+public class ExceptionUtils {
+
+    public static <T> Mono<T> handleWebClientResponseException(WebClientResponseException e, String prefixLog) {
+        var errorMessage = "%s, response status code %s, body returned '%s'".formatted(prefixLog, e.getStatusCode(), e.getResponseBodyAsString());
+        log.error(errorMessage, e);
+        return Mono.error(new ApiCallException(errorMessage, e));
+    }
+
+    public static <T> Mono<T> handleWebClientException(WebClientException e, String prefixLog) {
+        var errorMessage = "%s, and no response body returned".formatted(prefixLog);
+        log.error(errorMessage, e);
+        return Mono.error(new ApiCallException(errorMessage, e));
```

Comment:
lot code is duplicate in this function. I think these two function can be unified into one.